palantir-java-format/src/main/java/com/palantir/javaformat/java/JavaInputAstVisitor.java

```diff
@@ -3396,7 +3412,10 @@ int declareOne(
             if (modifiers.isPresent()) {
                 visitAndBreakModifiers(modifiers.get(), annotationsDirection, Optional.of(verticalAnnotationBreak));
             }
-            builder.open(type != null ? plusFour : ZERO);
+            boolean isVar = builder.peekToken().get().equals("var")
```

Comment:
ok so this is the first block where we diverge from the google-java-format version? https://github.com/google/google-java-format/commit/4ddb9148c103b9c65768e519884b0c8ad5caaf39#diff-0c133fd389bc6f207e146ba08dbc50dd78aa088ff31cdc56cb14ebb31e97ec49R158

Reply:
I put a debugger on this, and it _seems_ like the call stack of visitEnhancedForLoop -> visitToDeclare -> declareOne, we lost the `node` variable which could give us a first-class, typed way of detecting this rather than just relying on string comparisons.

I believe it would just be a matter of calling the `JCVariableDecl#isImplicitlyTyped()` method, and that would tell us we're looking at a `var` declaration. Not 100% sure if it's worth diverging from their implementation in order to plumb this down though, as it might make future maintenance more annoying, despite slightly cleaner code

Reply:
Happy to make the change but I would probably stick to the original impl if possible. That way we can cherry-pick new features and fixes instead of reimplementing them which will likely save us quite a bit of work down the road.